src/generic/graphicc.cpp

```diff
@@ -735,12 +737,35 @@ void wxCairoPenData::Init()
 wxCairoPenData::wxCairoPenData( wxGraphicsRenderer* renderer, const wxPen &pen )
     : wxCairoPenBrushBaseData(renderer, pen.GetColour(), pen.IsTransparent())
 {
+    wxDash *dashes;
+    int nb_dashes = pen.GetDashes(&dashes);
+    InitFromPenInfo(wxGraphicsPenInfo()
```

Comment:
It looks like we're doing this (constructing `wxGraphicsPenInfo` from an existing `wxPen`) in a couple of places, so I'd provide a `static wxGraphicsPenInfo CreateFromPen(const wxPen& pen)` method doing it and use it all of these places.